sklearn/__init__.py

```diff
@@ -95,6 +97,19 @@
                'clone', 'get_config', 'set_config', 'config_context',
                'show_versions']
 
+    if _openmp_warn_unsupported():
+        base_url = "dev" if __version__.endswith(".dev0") else "stable"
+        message = textwrap.dedent(
+            """
+            Scikit-learn has been built without OpenMP support. Some estimators
+            will run in sequential mode instead of leveraging thread-based
+            parallelism. You can find instructions to build scikit-learn with
+            OpenMP support at this adress:
+
+                https://scikit-learn.org/{}/developers/advanced_installation.html
+            """)
+        warnings.warn(message.format(base_url))
```

Comment:
Not sure this should be a UserWarning if there is another applicable category

Reply:
pandas has `pandas.errors.PerformanceWarning` maybe we could do something similar?